java-checks-test-sources/src/main/files/non-compiling/checks/RedundantTypeCastCheck.java

```diff
@@ -116,24 +116,49 @@ String qix2() {
   private byte value;
 
 
-  class FP_S1905 {
-    static class Overloaded {
-      static String f() {
-        return "";
-      }
-
-      static String f(String a) {
-        return "";
-      }
-    }
+}
 
-    void main() {
-      foo(Overloaded::f); // Does not compile without cast
-      foo(String::new); // Does not compile without cast
+class FP_S1905 {
+  static class Overloaded {
+    static String f() {
+      return "";
     }
 
-    void foo(Supplier<String> supplier) {
-    }
-    void foo(Function<String, String> function) {
+    static String f(String a) {
+      return "";
     }
-  }}
+  }
+
+  void main() {
+    foo(Overloaded::f); // Does not compile without cast
+    foo(String::new); // Does not compile without cast
+
+    bar(((Supplier<String>) Overloaded::fff));
```

Comment:
I would also put this test in the compiling part, to be sure that unknown method does not prevent the Exception to happen.